central/complianceoperator/v2/scanconfigurations/service/service_impl.go

```diff
@@ -55,7 +55,7 @@ var (
 		},
 	})
 
-	configNameRegexp = regexp.MustCompile(`^[a-z0-9][a-z0-9.-]*[a-z0-9]?$`)
+	configNameRegexp = regexp.MustCompile(`^[a-z0-9](?:[a-z0-9.-]*[a-z0-9])?$`)
```

Comment:
does this need to be a non capturing group if we are only matching the expression?

Reply:
I'd argue that since we aren't intending to capture anything it's more precise to make it a non-capturing group. The important part is that `[a-z0-9.-]*[a-z0-9]` needs to be in brackets so the "?" applies to the whole term.